internal/stoker/stokercr/scanners/trivy.go

```diff
@@ -0,0 +1,154 @@
+package scanners
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/aquasecurity/trivy/pkg/commands/artifact"
+	"github.com/aquasecurity/trivy/pkg/commands/operation"
+	"github.com/aquasecurity/trivy/pkg/db"
+	fanaltypes "github.com/aquasecurity/trivy/pkg/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/flag"
+	"github.com/aquasecurity/trivy/pkg/types"
+	"github.com/frantjc/sindri/internal/cache"
+	"github.com/frantjc/sindri/internal/stoker/stokercr"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/uuid"
+)
+
+type trivy struct {
+	runner   artifact.Runner
+	dbRepos  []name.Reference
+	cacheDir string
+}
+
+type TrivyOptions struct {
+	DBRepositories []string
+}
+
+type TrivyOption func(*TrivyOptions)
+
+func WithDBRepositories(repos []string) TrivyOption {
+	return func(o *TrivyOptions) {
+		o.DBRepositories = repos
+	}
+}
+
+func NewTrivy(ctx context.Context, opts ...TrivyOption) (*trivy, error) {
```

Comment:
As the lint suggests, the function signature should indicate the it returns the interface `controller.Scanner`, and then the function implementation can return `*trivy`